File: app/src/main/java/com/iomt/android/Note.java
package com.iomt.android;

import java.util.Map;

public class Note {
    public static final String TABLE_NAME = "hexoskin_data";

    public static final String COLUMN_ID = "id";
    public static final String COLUMN_HEART_RATE = "HeartRate";
    public static final String COLUMN_RESP_RATE = "RespRate";
    public static final String COLUMN_INSP = "Insp";
    public static final String COLUMN_EXP = "Exp";
    public static final String COLUMN_CADENCE = "Cadence";
    public static final String COLUMN_STEP_COUNT = "Steps";
    public static final String COLUMN_CLI = "Clitime";
    public static final String COLUMN_ACT = "Activity";
    public static final String COLUMN_MIL = "Millisec";
    public static final String COLUMN_TIMESTAMP = "timestamp";

    private int id;
    private Map<String, String> note;
    private String timestamp;


    // Create table SQL query
    public static final String CREATE_TABLE =
            "CREATE TABLE " + TABLE_NAME + "("
                    + COLUMN_ID + " INTEGER PRIMARY KEY AUTOINCREMENT,"
                    + COLUMN_HEART_RATE + " INTEGER,"
                    + COLUMN_RESP_RATE + " INTEGER,"
                    + COLUMN_INSP + " FLOAT,"
                    + COLUMN_EXP + " FLOAT,"
                    + COLUMN_CADENCE + " INTEGER,"
                    + COLUMN_STEP_COUNT + " INTEGER,"
                    + COLUMN_ACT + " FLOAT,"
                    + COLUMN_CLI + " TEXT,"
                    + COLUMN_MIL + " INTEGER,"
                    + COLUMN_TIMESTAMP + " DATETIME DEFAULT CURRENT_TIMESTAMP"
                    + ")";

    public Note() {
    }

    public Note(int id, Map<String, String> note, String timestamp) {
        this.id = id;
        this.note = note;
        this.timestamp = timestamp;
    }

    public int getId() {
        return id;
    }

    public Map<String, String> getNote() {
        return note;
    }

    public void setNote(Map<String, String> note) {
        this.note = note;
    }

    public String getTimestamp() {
        return timestamp;
    }

    public void setId(int id) {
        this.id = id;
    }

    public void setTimestamp(String timestamp) {
        this.timestamp = timestamp;
    }
}

File: app/src/main/java/com/iomt/android/Note.kt
package com.iomt.android

/**
 * @property id
 * @property note
 * @property timestamp
 */
@Suppress("MISSING_KDOC_CLASS_ELEMENTS")
data class Note(val id: Int, val note: Map<String, String>? = null, val timestamp: String? = null) {
    companion object {
        const val COLUMN_ACT = "Activity"
        const val COLUMN_CADENCE = "Cadence"
        const val COLUMN_CLI = "Clitime"
        const val COLUMN_EXP = "Exp"
        const val COLUMN_HEART_RATE = "HeartRate"
        const val COLUMN_ID = "id"
        const val COLUMN_INSP = "Insp"
        const val COLUMN_MIL = "Millisec"
        const val COLUMN_RESP_RATE = "RespRate"
        const val COLUMN_STEP_COUNT = "Steps"
        const val COLUMN_TIMESTAMP = "timestamp"

        const val TABLE_NAME = "hexoskin_data"

        /**
         * Get SQL create table query
         *
         * @param tableName name of a table in db, [TABLE_NAME] by default
         * @return SQL create table query
         */
        @Suppress("LONG_LINE")
        fun createTable(tableName: String = TABLE_NAME) = buildString {
            append("CREATE TABLE ")
            append(tableName)
            append("(")
            append(COLUMN_ID)
            append(" INTEGER PRIMARY KEY AUTOINCREMENT,")
            append(COLUMN_HEART_RATE)
            append(" INTEGER,")
            append(COLUMN_RESP_RATE)
            append(" INTEGER,")
            append(COLUMN_INSP)
            append(" FLOAT,")
            append(COLUMN_EXP)
            append(" FLOAT,")
            append(COLUMN_CADENCE)
            append(" INTEGER,")
            append(COLUMN_STEP_COUNT)
            append(" INTEGER,")
            append(COLUMN_ACT)
            append(" FLOAT,")
            append(COLUMN_CLI)
            append(" TEXT,")
            append(COLUMN_MIL)
            append(" INTEGER,")
            append(COLUMN_TIMESTAMP)
            append(" DATETIME DEFAULT CURRENT_TIMESTAMP)")
        }
    }
}
